Framework/Nexus/src/NexusFileIO.cpp

```diff
@@ -291,8 +292,11 @@ int NexusFileIO::writeNexusProcessedData2D(const API::MatrixWorkspace_const_sptr
   const size_t nHist = localworkspace->getNumberHistograms();
   if (nHist < 1)
     return (2);
-  const size_t nSpectBins = localworkspace->y(0).size();
   const size_t nSpect = indices.size();
+  size_t nSpectBins = localworkspace->y(0).size();
+  if (raggedSpectra)
+    for (size_t i = 0; i < nSpect; i++)
+      nSpectBins = std::max(nSpectBins, localworkspace->y(indices[i]).size());
```

Comment:
If you wanted to be fancy, you could use `std::accumulate` here... (and when finding `max_x_size` later)
```
size_t nSpectBins = std::accumulate(indicies.cbegin(), indicies.cend(), localworkspace->y(0).size(),
                                    [&localworkspace](const auto &currentMaxBins, const auto &index){
                                        return std::max(currentMaxBins, localworkspace->y(index).size());
                                    });
```